### PR TITLE
Docs/Examples - add optimistic-update example to repo

### PR DESCRIPTION
**Pull request by Khartir**
Adds the contents of the optimistic update example from the sandbox to the repo and updates the example to the current version of rtk-query.

### CODE REVIEW

File: .codesandbox/ci.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "sandboxes": [
     "vanilla",
     "vanilla-ts",
-    "github/reduxjs/rtk-github-issues-example"
+    "github/reduxjs/rtk-github-issues-example",
+    "github/reduxjs/toolkit/examples/query/react/optimistic-updates"
```
Comment (msutkowski) [inline .codesandbox/ci.json:6]:
```suggestion
```
I don't think we'd need all the examples for the docs built this way, but I could be missing something.

File: docs/rtk-query/usage/examples.mdx [hunk 2]
```diff
@@ -32,7 +32,7 @@ In the example below you'll notice a few things. There are two `Posts` list on t
 The example has some intentionally wonky behavior... when editing the name of a post, there is a decent chance you'll get a random error.
 :::
 
-<iframe src="https://codesandbox.io/embed/concepts-optimistic-updates-lush8?fontsize=12&hidenavigation=1&module=%2Fsrc%2Fapp%2Fservices%2Fposts.ts&theme=dark&runonclick=1"
+<iframe src="https://githubbox.com/reduxjs/redux-toolkit/tree/feature/v1.6-integration/examples/query/react/optimistic-update?fontsize=12&hidenavigation=1&module=%2Fsrc%2Fapp%2Fservices%2Fposts.ts&theme=dark&runonclick=1"
```
Comment (msutkowski) [inline docs/rtk-query/usage/examples.mdx:35]:
```suggestion
<iframe src="https://codesandbox.io/embed/github/reduxjs/redux-toolkit/tree/feature/v1.6-integration/examples/query/react/optimistic-update?fontsize=12&hidenavigation=1&module=%2Fsrc%2Fapp%2Fservices%2Fposts.ts&theme=dark&runonclick=1"
```

I don't see a problem with this, but long term we should probably just use the built-in CSB feature.

File: examples/query/react/optimistic-update/src/mocks/db.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+    const post = db.post.update({
+      where: { id: { equals: req.params.id } },
+      data: { name },
+    })
+
+    return res(ctx.json(post), ctx.delay(400))
+  }),
+  ...db.post.toHandlers('rest'),
+] as const
```
Comment (msutkowski) [inline examples/query/react/optimistic-update/src/mocks/db.ts:33]: :trophy: Nice!

**Review by msutkowski — APPROVED** [review]
LGTM, just a few comments, but not necessary to do now.

File: docs/rtk-query/usage/examples.mdx (hunk 2 above)
Comment (phryneas) [reply docs/rtk-query/usage/examples.mdx:35]: I thought so too, but apparently githubbox **is** a codesandbox feature. Just add `box` to any github link and you have a codesandbox

File: examples/query/react/optimistic-update/public/manifest.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "short_name": "Simple Query Demo",
+  "name": "RTK Simple Query Demo",
```
Comment (phryneas) [inline examples/query/react/optimistic-update/public/manifest.json:3]: Uhh we should change the internal CodeName here, even though it's kinda funny at this point :sweat_smile: 
```suggestion
  "short_name": "RTK Query Demo",
  "name": "RTK Query Demo",
```

**Comment by msutkowski** [comment]
Awesome! New to me 😅😅

**Review by phryneas — APPROVED** [review]

File: .codesandbox/ci.json (hunk 1 above)
Comment (phryneas) [reply .codesandbox/ci.json:6]: Can you specify?

File: docs/rtk-query/usage/examples.mdx (hunk 2 above)
Comment (Khartir) [reply docs/rtk-query/usage/examples.mdx]: I just realized that apparently https://githubbox.com cannot be embedded.

**Comment by phryneas** [comment]
Let's get this in :)

**Pull request merged**